### PR TITLE
CPLYTM-356 Initial documentation for the project

### PR DESCRIPTION
**Pull request by marcusburghardt**
This PR introduces an initial documentation for the project intended to make contributions easier.

For Style Guide it was used the best practices in Go based on my research, so they are basically formalization of already used practices.

For Code of Conducted I used an standard already used by many other projects and which seems reasonable, but I would like to hear other opinions.

For the Contributing it is very simple and generic. We can increment along the time.

### CODE REVIEW

File: docs/STYLE_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+## General Guidelines
+
+- **File Naming**: Use lowercase letters and underscores for file names (e.g., `my_file.go`).
+- **Package Names**: Use short, concise, and lowercase names for packages. Avoid underscores and mixed caps.
+- **Imports**: Group standard library imports separately from third-party imports and project-specific imports. Use blank lines to separate these groups.
+- **Error Handling**: Always check for errors and handle them appropriately. Return errors to the caller when necessary.
+- **Testing**: Write tests for your code. Use descriptive names for test functions and include edge cases.
+
+## Code Formatting
```
Comment (jpower432) [inline docs/STYLE_GUIDE.md:21]: Using `go fmt` should cover the indentation and spacing guides. I recommend we add some standard tools (e.g. `go fmt`, `go vet`, `goimports`) that developers should use for code consistency and have the style guide focus on aspects of the code that are more difficult to check with automation.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Contributing to ComplyTime
```
Comment (jpower432) [inline docs/CONTRIBUTING.md:1]: @marcusburghardt Was wondering if you think we should standardize on license header information in the source files here and provide guidance in the contributing guide. Maybe could just encourage the use of the SPDX identifier (`# SPDX-License-Identifier: Apache-2.0`)?

File: docs/STYLE_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Go Project Style Guide
+
+This style guide outlines the best practices to ensure consistency and readability across the codebase.
+
+## General Guidelines
+
+- **File Naming**: Use lowercase letters and underscores for file names (e.g., `my_file.go`).
+- **Package Names**: Use short, concise, and lowercase names for packages. Avoid underscores and mixed caps.
+- **Imports**: Group standard library imports separately from third-party imports and project-specific imports. Use blank lines to separate these groups.
```
Comment (huiwangredhat) [inline docs/STYLE_GUIDE.md]: We could use `goimports`, it will help ensure the imports are grouped correctly, with standard library imports first, followed by third-party imports, and finally project own imports; unused imports are removed; missing imports are added. It won't change the code logic, only the imports section.

File: docs/STYLE_GUIDE.md (hunk 1 above)
Comment (huiwangredhat) [reply docs/STYLE_GUIDE.md:21]: We could add the standard tools and golint to the CI/CD.

**Review by huiwangredhat — COMMENTED** [review]
LGTM. Just share some thoughts for the next CI with team.

File: docs/STYLE_GUIDE.md (hunk 1 above)
Comment (marcusburghardt) [reply docs/STYLE_GUIDE.md:21]: I will update this section to mention `go fmt` and `goimports`, which are more related to style. The `govet` was already introduced in CI/CD via https://github.com/complytime/complytime/pull/5 . I will include this information.

File: docs/STYLE_GUIDE.md (hunk 3 above)
Comment (marcusburghardt) [reply docs/STYLE_GUIDE.md]: Great. I saw it is already introduced in CI/CD via https://github.com/complytime/complytime/pull/5. I will reword this section.

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (marcusburghardt) [reply docs/CONTRIBUTING.md:1]: Nice. I think this is a good practice and adopting the SPDX format is a "clean" way to do that. I will include a sentence.

Comment (marcusburghardt) [reply docs/CONTRIBUTING.md:1]: Addressed by https://github.com/complytime/complytime/pull/2/commits/f2d3eb37b64245a699ba1fd73b58dea408fd755b

File: docs/STYLE_GUIDE.md (hunk 1 above)
Comment (marcusburghardt) [reply docs/STYLE_GUIDE.md:21]: Addressed by https://github.com/complytime/complytime/pull/2/commits/6a918a03b84911444ca795282590c7358254108c

File: docs/STYLE_GUIDE.md (hunk 3 above)
Comment (marcusburghardt) [reply docs/STYLE_GUIDE.md]: Addressed by https://github.com/complytime/complytime/pull/2/commits/6a918a03b84911444ca795282590c7358254108c

**Review by vojtapolasek — APPROVED** [review]
LGTM, thank you.

**Review by huiwangredhat — APPROVED** [review]

File: docs/STYLE_GUIDE.md (hunk 1 above)
Comment (huiwangredhat) [reply docs/STYLE_GUIDE.md:21]: The `goimports` is included in the golint in this [config file](https://github.com/complytime/complytime/blob/main/.golangci.yml)

**Review by jpower432 — APPROVED** [review]
LGTM
